test/compress/issue-1446.js

```diff
@@ -0,0 +1,75 @@
+typeof_eq_undefined: {
+    options = {
+        comparisons: true
+    }
+    input: {
+        var a = typeof b != "undefined";
+        b = typeof a != "undefined";
+        var c = typeof d.e !== "undefined";
+        var f = "undefined" === typeof g;
+        g = "undefined" === typeof f;
+        var h = "undefined" == typeof i.j;
+    }
+    expect: {
+        var a = "undefined" != typeof b;
+        b = void 0 !== a;
+        var c = void 0 !== d.e;
+        var f = "undefined" == typeof g;
+        g = void 0 === f;
+        var h = void 0 === i.j;
+    }
+}
+
+typeof_eq_undefined_ie8: {
+    options = {
+        comparisons: true,
+        screw_ie8: false
+    }
+    input: {
+        var a = typeof b != "undefined";
+        b = typeof a != "undefined";
+        var c = typeof d.e !== "undefined";
+        var f = "undefined" === typeof g;
+        g = "undefined" === typeof f;
+        var h = "undefined" == typeof i.j;
+    }
+    expect: {
+        var a = "undefined" != typeof b;
+        b = void 0 !== a;
+        var c = "undefined" != typeof d.e;
+        var f = "undefined" == typeof g;
+        g = void 0 === f;
+        var h = "undefined" == typeof i.j;
+    }
+}
+
+undefined_redefined: {
+    options = {
+        comparisons: true
+    }
+    input: {
+        function f(undefined) {
+            return function() {
+                var n = 1;
+                return typeof n == "undefined";
+            };
+        }
+    }
+    expect_exact: "function f(undefined){return function(){var n=1;return void 0===n}}"
+}
+
+undefined_redefined_mangle: {
+    options = {
+        comparisons: true
+    }
+    mangle = {}
+    input: {
+        function f(undefined) {
+            return function() {
```

Comment:
Please remove the superfluous inner function so that `undefined` and `var n` are in the same scope.